DiscountDiscover/View Controllers/SignUpViewController.m

```diff
@@ -0,0 +1,75 @@
+//
+//  SignUpViewController.m
+//  DiscountDiscover
+//
+//  Created by Kaitlyn Gunadhi on 7/23/20.
+//  Copyright © 2020 Kaitlyn Gunadhi. All rights reserved.
+//
+
+#import "SignUpViewController.h"
+#import <Parse/Parse.h>
+#import "LoginViewController.h"
+#import "UIViewController+Error.h"
+
+@interface SignUpViewController ()
+
+@property (weak, nonatomic) IBOutlet UITextField *emailField;
+@property (weak, nonatomic) IBOutlet UITextField *nameField;
+@property (weak, nonatomic) IBOutlet UITextField *usernameField;
+@property (weak, nonatomic) IBOutlet UITextField *passwordField;
+@property (weak, nonatomic) IBOutlet UIButton *signupButton;
+
+@end
+
+@implementation SignUpViewController
+
+- (void)viewDidLoad {
+    [super viewDidLoad];
+    
+    self.signupButton.layer.cornerRadius = 7;
+    self.signupButton.clipsToBounds = YES;
+}
+
+- (IBAction)onTap:(id)sender {
+    [self.view endEditing:YES];
+}
+
+- (IBAction)registerUser:(id)sender {
+    // initialize a user object
+    PFUser *newUser = [PFUser user];
+    
+    // set user properties
+    newUser.email = self.emailField.text;
+    newUser[@"name"] = self.nameField.text;
```

Comment:
You should always use the . notation instead of the subscript notation when accessing model properties, in this case `newUser.name` instead of `newUser[@"name"]`, so the compiler can help you find any issues early on and avoid crashes or bugs at runtime.